pyrefinebio/compendia.py

```diff
@@ -95,6 +98,29 @@ def download(self, path, prompt=True):
 
             prompt (bool): if true, will prompt before downloading files bigger than 1GB
         """
-        self.computed_file.download(path, prompt)
+        if not self.computed_file.download_url:
+            raise DownloadError("Compendia", "Download url not found - did you set up and activate a Token?")
+
+        full_path = expand_path(path, "compendia-" + str(self.id) + ".zip")
+
+        download_file(self.computed_file.download_url, full_path, prompt)
+
+        self._downloaded_path = full_path
 
         return self
+
+
+    def extract(self):
+        """Extract a downloaded Compendium
+
+        Returns:
+            Compendium
+        """
+        if not self._downloaded_path:
+            raise Exception(
```

Comment:
Can we use a less generic exception class?

Reply:
I wasn't sure which exception to use.
maybe `DownloadError` or should I make a new class for extraction related errors?

Reply:
Mmm maybe a new class like MissingFile or something?

Reply:
added MissingFile exception class